### PR TITLE
unify leader change and machine not found

### PR DESCRIPTION
**Pull request by darionyaphet**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [ ] enhancement

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Review by HarrisChu — APPROVED** [review]

**Review by Aiee — APPROVED** [review]

File: src/clients/meta/MetaClient.cpp [hunk 1]
```diff
@@ -743,7 +743,8 @@ void MetaClient::getResponse(Request req,
                 // succeeded
                 pro.setValue(respGen(std::move(resp)));
                 return;
-              } else if (code == nebula::cpp2::ErrorCode::E_LEADER_CHANGED) {
+              } else if (code == nebula::cpp2::ErrorCode::E_LEADER_CHANGED ||
+                         code == nebula::cpp2::ErrorCode::E_MACHINE_NOT_FOUND) {
```
Comment (critical27) [inline src/clients/meta/MetaClient.cpp:747]: If so, why not just handle only return `E_MACHINE_NOT_FOUND` when the key not in `ActiveHostsMan::machineRegisted`...

**Comment by HarrisChu** [comment]
BTW, prefer to handle error code in heart beat.

```
    if (!ActiveHostsMan::machineRegisted(kvstore_, host)) {
      # return leader change code if meta is follower


```

**Comment by darionyaphet** [comment]
> BTW, prefer to handle error code in heart beat.
> 
> ```
>     if (!ActiveHostsMan::machineRegisted(kvstore_, host)) {
>       # return leader change code if meta is follower
> ```

`setLeaderInfo` have processed this problem

File: src/clients/meta/MetaClient.cpp (hunk 1 above)
Comment (darionyaphet) [reply src/clients/meta/MetaClient.cpp:747]: when the active hosts manager not found the hosts return `E_MACHINE_NOT_FOUND` and set leader message

Comment (critical27) [reply src/clients/meta/MetaClient.cpp:747]: I mean only leader could return `E_MACHINE_NOT_FOUND`, follower will return LEADER_CHANGED, so we don't need to have to add code in `MetaClient`

**Comment by critical27** [comment]
Merge first, the code could be improved later

**Pull request merged**